test/interfaces/flags.test.ts

```diff
@@ -0,0 +1,249 @@
+    defaultMultiEnum: Flags.enum({multiple: true, options: ['a', 'b', 'c'], default: ['a']}),
+
+    optionalTypedEnum: Flags.enum<MyEnum>({options: Object.values(MyEnum)}),
+    requiredTypedEnum: Flags.enum<MyEnum>({options: Object.values(MyEnum), required: true}),
+    defaultTypedEnum: Flags.enum<MyEnum>({options: Object.values(MyEnum), default: MyEnum.A}),
+
+    optionalInteger: Flags.integer(),
+    requiredInteger: Flags.integer({required: true}),
+    defaultInteger: Flags.integer({default: 1}),
+
+    optionalMultiInteger: Flags.integer({multiple: true}),
+    requiredMultiInteger: Flags.integer({multiple: true, required: true}),
+    defaultMultiInteger: Flags.integer({multiple: true, default: [1]}),
+
+    optionalDirectory: Flags.directory(),
+    requiredDirectory: Flags.directory({required: true}),
+    defaultDirectory: Flags.directory({default: 'my-dir'}),
+
+    optionalMultiDirectory: Flags.directory({multiple: true}),
+    requiredMultiDirectory: Flags.directory({multiple: true, required: true}),
+    defaultMultiDirectory: Flags.directory({multiple: true, default: ['my-dir']}),
+
+    optionalFile: Flags.file(),
+    requiredFile: Flags.file({required: true}),
+    defaultFile: Flags.file({default: 'my-file.json'}),
+
+    optionalMultiFile: Flags.file({multiple: true}),
+    requiredMultiFile: Flags.file({multiple: true, required: true}),
+    defaultMultiFile: Flags.file({multiple: true, default: ['my-file.json']}),
+
+    optionalUrl: Flags.url(),
+    requiredUrl: Flags.url({required: true}),
+    defaultUrl: Flags.url({default: new URL('http://example.com')}),
+
+    optionalMultiUrl: Flags.url({multiple: true}),
+    requiredMultiUrl: Flags.url({multiple: true, required: true}),
+    defaultMultiUrl: Flags.url({multiple: true, default: [new URL('http://example.com')]}),
+
+    optionalBuild: Flags.build<MyType>({
+      parse: async () => ({foo: true}),
+    })(),
+    requiredBuild: Flags.build<MyType>({
+      parse: async () => ({foo: true}),
+    })({required: true}),
+    defaultBuild: Flags.build<MyType>({
+      parse: async () => ({foo: true}),
+    })({default: {foo: true}}),
+
+    optionalCustom: Flags.custom<MyType>({
+      parse: async () => ({foo: true}),
+    })(),
+    requiredCustom: Flags.custom<MyType>({
+      parse: async () => ({foo: true}),
+    })({required: true}),
+    defaultCustom: Flags.custom<MyType>({
+      parse: async () => ({foo: true}),
+    })({default: {foo: true}}),
+
+    optionalMultiCustom: Flags.custom<MyType>({
+      parse: async () => ({foo: true}),
+    })({multiple: true}),
+    requiredMultiCustom: Flags.custom<MyType>({
+      parse: async () => ({foo: true}),
+    })({required: true, multiple: true}),
+    defaultMultiCustom: Flags.custom<MyType>({
+      parse: async () => ({foo: true}),
+    })({default: [{foo: true}], multiple: true}),
+  }
+
+  public flags!: MyFlags
+
+  public async run(): Promise<MyFlags> {
+    const result = await this.parse(MyCommand)
```

Comment:
I understand why the `!` is there in here and in your example.  It's kinda true...as long as people make 
`const {flags} = await this.parse(MyCommand)` (or whatever) their very first step in `run`.

Is that worth adding to our linter?  Are there situations where someone wouldn't want to parse the flags first (ex: checking some env, etc?) or not at all?

Otherwise, if they do the `public flags!: MyFlags` but forget to parse (or try to use this.flags before parsing) the compiler won't warn them.

Or maybe the rule is "don't refer to the instance `flags` prop if you haven't called the parse and assigned it?" ?